### PR TITLE
tidb-lightning: add line terminator and update other csv settings

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:  https://github.com/pingcap/docs/pull/6386
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 1]
```diff
@@ -30,10 +30,13 @@ CSV 格式可在 `tidb-lightning.toml` 文件中 `[mydumper.csv]` 下配置。
...
 ```toml
 [mydumper.csv]
-# 字段分隔符，必须为 ASCII 字符。如果数据中可能有逗号，建议源文件导出时分隔符使用非常见组合字符例如'|+|'。
+# 字段分隔符，支持一个或多个字符，默认值为 ','。如果数据中可能有逗号，建议源文件导出时分隔符使用非常见组合字符例如'|+|'。
 separator = ','
-# 引用定界符，可以为 ASCII 字符或空字符。
+# 引用定界符，设置为空表示字符串未加引号。
 delimiter = '"'
+# 行尾定界字符，支持一个或多个字符。设置为空（默认值）表示 "\n"（换行）和 "\r\n" （回车+换行）均表示行尾。
+# 注意：如果同时设置非空的 terminator 并且 trim-last-separator 为 true，则 trim-last-separator 将被忽略。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:38]: 这句话英文没有。辛苦@hfxsd 再提个英文 PR 在英文里加上这句话吧~

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 2]
```diff
@@ -45,20 +48,27 @@ not-null = false
 null = '\N'
 # 是否解析字段内的反斜线转义符。
 backslash-escape = true
-# 是否移除以分隔符结束的行。
+# 是否移除以分隔符结束的行。只在 terminator 配置为空时生效。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:51]: 这里英文也需要加

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 3]
```diff
@@ -135,7 +166,6 @@ TiDB Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例
 * 行终止符只能是 CR（`\r`），LF（`\n`）或 CRLF（`\r\n`），也就是说，无法自定义 `LINES TERMINATED BY`。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:166]:
```suggestion
```
这个我看英文的删除了呢。@glorv

File: tidb-lightning/tidb-lightning-configuration.md [hunk 4]
```diff
@@ -163,14 +163,17 @@ strict-format = false
...
 # 配置 CSV 文件的解析方式。
 [mydumper.csv]
-# 字段分隔符，应为单个 ASCII 字符。
+# 字段分隔符，支持一个或多个字符，默认值为 ','。
 separator = ','
-# 引用定界符，可为单个 ASCII 字符或空字符串。
+# 引用定界符，设置为空表示字符串未加引号。
 delimiter = '"'
+# 行尾定界字符，支持一个或多个字符。设置为空（默认值）表示 "\n"（换行）和 "\r\n" （回车+换行）均表示行尾。
+# 注意：如果同时设置非空的 terminator 并且 trim-last-separator 为 true，则 trim-last-separator 将被忽略。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-configuration.md:175]: 这里英文也需要加上。@hfxsd

**Review by Liuxiaozhen12 — COMMENTED** [review]
rest LGTM

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md (hunk 1 above)
Comment (kennytm) [reply tidb-lightning/migrate-from-csv-using-tidb-lightning.md:38]: There is no such behavior. Setting `terminator` won't disable `trim-last-separator`.

Comment (glorv) [reply tidb-lightning/migrate-from-csv-using-tidb-lightning.md:38]: yes, I recheck the current logic and found I misunderstood it，😂.
BTW, I'm not sure if there are any cases that user need to set both `terminator` and `trim-last-seqarator`.

File: tidb-lightning/tidb-lightning-configuration.md [hunk 5]
```diff
@@ -163,14 +163,16 @@ strict-format = false
...
 
 # 配置 CSV 文件的解析方式。
 [mydumper.csv]
-# 字段分隔符，应为单个 ASCII 字符。
+# 字段分隔符，支持一个或多个字符，默认值为 ','。
 separator = ','
-# 引用定界符，可为单个 ASCII 字符或空字符串。
+# 引用定界符，设置为空表示字符串未加引号。
 delimiter = '"'
+# 行尾定界字符，支持一个或多个字符。设置为空（默认值）表示 "\n"（换行）和 "\r\n" （回车+换行）均表示行尾。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-configuration.md:174]:
```suggestion
# 行尾定界字符，支持一个或多个字符。设置为空（默认值）则表示 "\n"（换行）和 "\r\n" （回车+换行），均表示行尾。
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 6]
```diff
@@ -75,6 +84,12 @@ trim-last-separator = false
 
 [RFC 4180]: https://tools.ietf.org/html/rfc4180
 
+### `terminator`
+
+- 指定行尾定界符。
+- 如果 `terminator` 为空，表示 "\n"（换行）和 "\r\n" （回车+换行）均表示行尾。
```
Comment (hfxsd) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:90]:
```suggestion
- 如果 `terminator` 为空，则表示 "\n"（换行）和 "\r\n" （回车+换行），均表示行尾。
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 7]
```diff
@@ -30,10 +30,12 @@ CSV 格式可在 `tidb-lightning.toml` 文件中 `[mydumper.csv]` 下配置。
 
 ```toml
 [mydumper.csv]
-# 字段分隔符，必须为 ASCII 字符。如果数据中可能有逗号，建议源文件导出时分隔符使用非常见组合字符例如'|+|'。
+# 字段分隔符，支持一个或多个字符，默认值为 ','。如果数据中可能有逗号，建议源文件导出时分隔符使用非常见组合字符例如'|+|'。
 separator = ','
-# 引用定界符，可以为 ASCII 字符或空字符。
+# 引用定界符，设置为空表示字符串未加引号。
 delimiter = '"'
+# 行尾定界字符，支持一个或多个字符。设置为空（默认值）表示 "\n"（换行）和 "\r\n" （回车+换行）均表示行尾。
```
Comment (hfxsd) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:37]:
```suggestion
# 行尾定界字符，支持一个或多个字符。设置为空（默认值）则表示 "\n"（换行）和 "\r\n" （回车+换行），均表示行尾。
```

**Review by hfxsd — APPROVED** [review]

**Review by kennytm — APPROVED** [review]

**Comment by glorv** [comment]
v4.0, v5.0, v5.1 support multi character separator/delimiter but not line-terminator. 😂

**Comment by glorv** [comment]
@hfxsd shall we merge is PR now?

**Pull request merged**